gensim/summarization/bm25.py

```diff
@@ -151,8 +173,8 @@ def get_score(self, document, index):
         for word in document:
             if word not in doc_freqs:
                 continue
-            score += (self.idf[word] * doc_freqs[word] * (PARAM_K1 + 1)
-                      / (doc_freqs[word] + PARAM_K1 * (1 - PARAM_B + PARAM_B * self.doc_len[index] / self.avgdl)))
+            score += (self.idf[word] * doc_freqs[word] * (self.k1 + 1)
```

Comment:
Now that we have some tests, I think this PR is a good opportunity to make this more readable.

How about this inside the loop:

```python
# Maybe pick a better name for this? Looks like we can totally precompute it.
numerator_pad = self.k1 + 1
denominator_pad = self.k1 * (1 - self.b + self.b * self.doc_len[index] / self.avgd1)
for word in document:
    idf = self.idf[word]
    try:
        df = self.doc_freqs[index][word]
    except KeyError:
        # ask for forgiveness instead of permission
        continue
    score +=  (idf * df * numerator_pad) / (df + denominator_pad)
return score
```

WDYT?

Reply:
`word not in doc_freqs:` much more readable than catching exceptions, I prefer the original.

In general, is the division guaranteed not to raise division by zero?

Reply:
OK, I updated my suggestion.

Yeah, I think we're OK with the div-by-zero because of the denominator_pad. @Witiko Will that always be non-zero?

Reply:
"Not equal to `-df`" is what we need.

Reply:
I am skimming the code, but, as long as `b` is within [0; 1] and `k1` is non-negative, `denominator_pad` is non-negative and `df` is positive, so no division by zero will occur.

If `corpus` is empty, then `self.avgdl` is zero, which would seem to lead to a division by zero, but the user will receive an `IndexError` before that.

Should I break down the BM25 formula into subcomponents like @mpenkov suggests?

Reply:
I refactored the inner loop in 13845d12.